library/src/main/java/org/apache/apex/malhar/lib/utils/serde/SerdeListSlice.java

```diff
@@ -38,11 +38,12 @@
 public class SerdeListSlice<T> implements Serde<List<T>, Slice>
 {
   @NotNull
-  private Serde<T, Slice> serde;
+  protected Serde<T, Slice> serde;
 
-  private SerdeListSlice()
+  protected SerdeListSlice()
   {
     // for Kryo
+    throw new RuntimeException("should only called by Kryo");
```

Comment:
And when it is called it should fail? In that case it is not needed and can be removed?